frontend/src/components/housing/maps/Map.vue

```diff
@@ -0,0 +1,89 @@
+<script setup lang="ts">
+import * as L from 'leaflet';
+import 'leaflet/dist/leaflet.css';
+import '@geoman-io/leaflet-geoman-free';
+import '@geoman-io/leaflet-geoman-free/dist/leaflet-geoman.css';
+
+import {
+  BC_BOUNDARIES_LOWER,
+  BC_BOUNDARIES_UPPER,
+  MAP_ICON_OPTIONS_RED,
+  MAP_INITIAL_START_POINT,
+  OSM_TILE_LAYER_OPTIONS,
+  OSM_URL_TEMPLATE
+} from '@/utils/constants/housing';
+import { onMounted, onUpdated } from 'vue';
+
+type Props = {
+  latitude?: number;
+  longitude?: number;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  latitude: undefined,
+  longitude: undefined
+});
+
+// Actions
+let marker: L.Marker;
+let map: L.Map;
+
+// Map component will be misaligned if mounted while not visible. Trigger resize to fix on show
+// Parent component needs to trigger nextTick().then(() => mapRef?.value?.resizeMap()
+function resizeMap() {
+  if (map) map.invalidateSize();
+}
+
+function setAddressMarker(coords: any) {
+  if (marker) map.removeLayer(marker);
+  // Custom(-ish) markers courtesy of https://github.com/pointhi/leaflet-color-markers
+  const redIcon = new L.Icon(MAP_ICON_OPTIONS_RED);
+  marker = L.marker(coords, { icon: redIcon });
+  map.addLayer(marker);
+}
+
+async function initMap() {
+  const osm = L.tileLayer(OSM_URL_TEMPLATE, OSM_TILE_LAYER_OPTIONS);
+
+  map = L.map('map', {
+    ...MAP_INITIAL_START_POINT,
+    layers: [osm]
+  });
+
+  // Don't allow dragging outside BC
+  const bcBounds = new L.LatLngBounds(BC_BOUNDARIES_LOWER, BC_BOUNDARIES_UPPER);
+  map.setMaxBounds(bcBounds);
+  map.on('drag', function () {
+    map.panInsideBounds(bcBounds, { animate: false });
+  });
+}
+
+onMounted(async () => {
+  await initMap();
+});
+
+defineExpose({ resizeMap });
```

Comment:
We haven't used these yet, but having it sit between lifecycle functions doesn't feel like the best spot to me?